Runtime/Core/ShrinkablePolygon.cs

```diff
@@ -0,0 +1,1102 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Cinemachine.Utility;
+using ClipperLib;
+using UnityEngine;
+using UnityEngine.Assertions;
+
+namespace Cinemachine
+{
+    /// <summary>
+    /// ShrinkablePolygon represents points with shrink directions.
+    /// </summary>
+    internal class ShrinkablePolygon
+    {
+        /// <summary>
+        /// 2D point with shrink direction.
+        /// </summary>
+        public struct ShrinkablePoint2
+        {
+            public Vector2 m_Position;
+            public Vector2 m_OriginalPosition;
+            public Vector2 m_ShrinkDirection;
+            public bool m_CantIntersect;
+            public static readonly Vector2 m_Vector2NaN = new Vector2(float.NaN, float.NaN);
+
+            // public ShrinkablePoint2()
+            // {
+            //     m_OriginalPosition = m_Vector2NaN;
+            // }
+            //
+            // public ShrinkablePoint2(Vector2 mPosition, Vector2 mOriginalPosition, Vector2 mShrinkDirection, 
+            //     bool mCantIntersect)
+            // {
+            //     m_Position = mPosition;
+            //     m_OriginalPosition = mOriginalPosition;
+            //     m_ShrinkDirection = mShrinkDirection;
+            //     m_CantIntersect = mCantIntersect;
+            // }
+        }
+
+        public List<ShrinkablePoint2> m_Points;
+        public float m_WindowDiagonal;
+        public int m_State;
+        public readonly float m_AspectRatio;
+        public readonly float m_AspectRatioBasedDiagonal;
+        public readonly Vector2[] m_NormalDirections;
```

Comment:
Will the m_NormalDirections be the same for all shrinkable polygons in the oven?

Reply:
Yes. They depend on the aspect ratio only.